### PR TITLE
[FEATURE] Added wildcard selector support. Fixes #303

### LINKED ISSUE #303: Add support for wildcards in attribute selectors
Hello, 

currently it is not possible to use a wildcard selector like `[class*=foobar]`. 

**Example (wrong xpath result):**
```
CSS: .search-container [class*=foobar]
Xpath: //*[contains(concat(" ",@class," "),concat(" ","search-container"," "))]//[class*=foobar]
```

Therefore i added another element to the `$xPathRules` array which will convert the css selector into a correct xpath expression: 

```
'/\\[(\\w+)\\*=([\\w-]+)\\]/'               => '*[contains(concat(" ",@\\1," "),"\\2")]',
```

**Example (correct xpath result):**
```
CSS: .search-container [class*=foobar]
Xpath: //*[contains(concat(" ",@class," "),concat(" ","search-container"," "))]//*[contains(concat(" ",@class," "),"foobar")]
```

I will make a pull request for this feature.

### PR DESCRIPTION
**Pull request by ernsto**

### CODE REVIEW

**Comment by oliverklee** [comment]
Hi @ernsto, thanks for the PR! I'll review it tomorrow.

In the meantime, could you please move the new entry in the changelog to the top of the "Added" section so that the newest entry comes first? Thanks!

File: Tests/Unit/EmogrifierTest.php [hunk 1]
```diff
@@ -609,6 +609,30 @@ public function selectorDataProvider()
...
+            'wildcard selector [class*=P-3] not matches second P elements'    =>
+                ['[class*=p-3] {' . $styleRule . '} ', '#<p class="p-2">#'],
+            'wildcard selector [class*=P-3] matches third P elements'    =>
+                ['[class*=p-3] {' . $styleRule . '} ', '#<p class="p-3" ' . $styleAttribute . '>#'],
+            'wildcard selector [class*=P-4] not matches first P element'    =>
+                ['[class*=p-4] {' . $styleRule . '} ', '#<p class="p-2">#'],
+            'wildcard selector [class*=P-4] not matches second P element'    =>
+                ['[class*=p-4] {' . $styleRule . '} ', '#<p class="p-2">#'],
+            'wildcard selector [class*=P-4] not matches third P element'    =>
+                ['[class*=p-4] {' . $styleRule . '} ', '#<p class="p-3">#'],
```
Comment (oliverklee) [inline Tests/Unit/EmogrifierTest.php]: Generally, it's great that you've covered this with tests. I think we still can improve the tests.

For the wildcard selector, it's not important whether it's the first, second etc. P tag. Instead, I propose the following cases:

positive:
* selector value is an exact match with the attribute value
* selector value is a prefix of the attribute value
* selector value is a suffix of the the attribute value
* selector value is in the middle of the attribute value

negative cases:
* completely different strings
* non-empty selector value, empty attribute value
* non-empty selector value, missing attribute value
* empty selector value, non-empty attribute value (what's the expected behavior here? Is this even valid?)

What do you think?

Comment (ernsto) [inline Tests/Unit/EmogrifierTest.php]: I think you are right. Your test cases are better. Thank you for your instructions. I will implement the new test cases in the future.

Comment (oliverklee) [inline Tests/Unit/EmogrifierTest.php]: Could you please post here in the PR when you've updated the test cases? Then I'll re-review the PR. Thanks!

**Comment by oliverklee** [comment]
@ernsto Any news on this PR?

**Comment by ernsto** [comment]
Hi. Sorry for the late answer. I had no time to change the unit tests. Is it possible that someone else change them? I think i won't have time for this in the near future.

**Comment by oliverklee** [comment]
Hi @ernsto: This ticket will most probably only go forward if you update the unit tests yourself. Only you know what needs to be tested for your changes. (And I am willing to give pull request a final small polish, but not completely overhaul it or write the unit tests.)

**Comment by oliverklee** [comment]
Closing this PR as I won't be able to merge it in its current state. @ernsto Could you please reopen this PR once you've done the necessary changes and re-pushed? Thanks!

**Pull request closed without merging**